# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLCanvasElement.cpp
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/html/HTMLCanvasElement.cpp
--- a/dom/html/HTMLCanvasElement.cpp	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/html/HTMLCanvasElement.cpp	Wed Jan 08 22:19:16 2020 +0000
@@ -1354,7 +1354,7 @@
 
 layers::OOPCanvasRenderer* HTMLCanvasElement::GetOOPCanvasRenderer() {
   if (!mOOPCanvasRenderer) {
-    ClientWebGLContext* context = GetClientWebGLContext();
+    const auto context = GetWebGLContext();
     MOZ_ASSERT(context);
     mOOPCanvasRenderer = new OOPCanvasRenderer(context);
   }
@@ -1487,7 +1487,7 @@
   element->InvalidateCanvasContent(nullptr);
 }
 
-ClientWebGLContext* HTMLCanvasElement::GetClientWebGLContext() {
+ClientWebGLContext* HTMLCanvasElement::GetWebGLContext() {
   if (GetCurrentContextType() != CanvasContextType::WebGL1 &&
       GetCurrentContextType() != CanvasContextType::WebGL2) {
     return nullptr;
@@ -1496,28 +1496,5 @@
   return static_cast<ClientWebGLContext*>(GetContextAtIndex(0));
 }
 
-PWebGLChild* HTMLCanvasElement::GetWebGLChild() {
-  ClientWebGLContext* webgl = GetClientWebGLContext();
-  if (!webgl) {
-    return nullptr;
-  }
-
-  return webgl->GetWebGLChild();
-}
-
-void HTMLCanvasElement::ClearVRFrame() {
-  if (GetCurrentContextType() != CanvasContextType::WebGL1 &&
-      GetCurrentContextType() != CanvasContextType::WebGL2) {
-    return;
-  }
-
-  WebGLContext* webgl = static_cast<WebGLContext*>(GetContextAtIndex(0));
-  if (!webgl) {
-    return;
-  }
-
-  webgl->ClearVRFrame();
-}
-
 }  // namespace dom
 }  // namespace mozilla